miio/curtain_babai.py

```diff
@@ -0,0 +1,201 @@
+        """Retrieve properties."""
+        # if use get_properties_for_mapping and max_len>1 then code -4004.
+        data = []
+        for k, v in self.mapping.items():
+            if "aiid" not in v:
+                data.extend(self.get_property(k))
+                # data[k] = prop["value"] if prop["code"] == 0 else None
+        return CurtainStatus(self._extract_property_value(data))
+
+    def get_property(self, name: str) -> List[Dict[str, Any]]:
+        if name not in self.mapping:
+            raise ValueError("key %s not in MAPING", name)
+        v = self.mapping[name]
+        try:
+            prop = self.get_property_by(siid=v["siid"], piid=v["piid"])
+        except DeviceError as e:
+            if e.code != -9999:
+                raise
+            sleep(5)
+            prop = self.get_property_by(siid=v["siid"], piid=v["piid"])
+        for p in prop:
+            p["did"] = name
+        return prop
+
+    @staticmethod
+    def _extract_property_value(prop: List[Dict[str, Any]]) -> Dict[str, Any]:
+        return {p["did"]: p["value"] if p["code"] == 0 else None for p in prop}
+
+    def _get_property_value(self, name: str) -> Any:
+        prop = self.get_property(name)
+        return self._extract_property_value(prop).get(name)
+
+    @property
+    def motor_control(self) -> MotorControl:
+        """Device status."""
+        return MotorControl(self._get_property_value("motor_control"))
+
+    @property
+    def mode(self) -> Mode:
+        """Motor rotation polarity."""
+        return Mode(self._get_property_value("mode"))
+
+    @property
+    def current_position(self) -> int:
+        """Run time of the motor."""
+        return self._get_property_value("current_position")
+
+    @property
+    def target_position(self) -> int:
+        """Target curtain position."""
+        return self._get_property_value("target_position")
+
+    @command(
+        click.argument("motor_control", type=EnumType(MotorControl)),
+        default_output=format_output("Set motor control to {motor_control}"),
+    )
+    def set_motor_control(self, motor_control: MotorControl):
+        """Set motor control."""
+        return self.set_property("motor_control", motor_control.value)
+
+    @command(
+        click.argument("open"),
+        default_output=format_output("Open curtain"),
+    )
+    def set_open(self):
+        """Set motor control."""
+        return self.set_motor_control(MotorControl.Open)
+
+    @command(
+        click.argument("close"),
+        default_output=format_output("Close curtain"),
+    )
+    def set_close(self):
+        """Set motor control."""
+        return self.set_motor_control(MotorControl.Close)
+
+    @property
+    def is_open(self) -> bool:
+        """Device status."""
+        return self.motor_control == MotorControl.Open
+
+    @property
+    def is_close(self) -> bool:
+        """Device status."""
+        return self.motor_control == MotorControl.Close
```

Comment:
No need for these here, the container returned by `status()` should contain all the necessary status information (and accessors for those).

Reply:
status not lasy ^(